docs/source/api.rst

```diff
@@ -243,6 +243,52 @@ Chalice
         entire lambda function name.  This parameter is optional.  If it is
         not provided, the name of the python function will be used.
 
+   .. method:: on_sqs_message(queue, batch_size=1, name=None)
+
+      Create a lambda function and configure it to be automatically invoked
+      whenever a message is published to the specified SQS queue.
+
+      The lambda function must accept a single parameter which
+      is of type :class:`SQSEvent`.
+
+      If the decorated function returns without raising any exceptions
+      then Lambda will automatically delete the SQS messages associated
+      with the :class:`SQSEvent`.  You don't need to manually delete
+      messages.  If any exception is raised, Lambda won't delete any messages,
+      and the messages will become available once the visibility timeout
+      has been reached.  Note that for batch sizes of more than one, either
+      the entire batch succeeds and all the messages in the batch are
+      deleted by Lambda, or the entire batch fails.  See the
+      `Using AWS Lambda with Amazon SQS <https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html>`__
+      for more information on how Lambda integrates with SQS.
+
+      See the :ref:`sqs-events` topic guide for more information on using SQS
+      in Chalice.
+
+      .. code-block:: python
+
+          app.debug = True
+
+          @app.on_sqs_message(queue='myqueue')
+          def handler(event):
+              app.log.info("Event: %s", event.to_dict())
+              for record in event:
+                  app.log.info("Message body: %s", record.body)
+
+      :param queue: The name of the SQS queue you want to subscribe to.
+        This is the name of the queue, not the ARN or Queue URL.
+
+      :param batch_size: The maximum number of messages to retrieve
+        when polling for SQS messages.  The event parameter can have
+        multiple SQS messages associated with it.  This is why the
+        event parameter passed to the lambda function is iterable.  The
+        batch size controls how many messages can be in a single event.
```

Comment:
It's in the function signature above but it might be worth mentioning that we default this to 1 here as well.

Reply:
Pushed a change for this.